src/shared/Specimen.ts

```diff
@@ -0,0 +1,100 @@
+import type {VisualizerInterface} from '../visualizers/VisualizerInterface'
+import type {SequenceInterface} from '../sequences/SequenceInterface'
+
+/**
+ * This class represents a specimen, containing a visualizer,
+ * a sequence, and the set of all parameters that make both of them up.
+ * Specimens can be converted to and from URLs so that they can be saved.
+ */
+export class Specimen {
+    private visualizer: VisualizerInterface
+    private sequence: SequenceInterface
+    private location?: HTMLElement
+    private isSetup: boolean = false
+
+    /**
+     * Constructs a new specimen from a visualizer, and a sequence
+     * It is not important that the visualizer
+     * is constructed with the same sequence as the one passed to this
+     * constructor, because the visualizer will be automatically set
+     * to view the specimen's visualizer in this constructor.
+     * @param visualizer the specimen's visualizer
+     * @param sequence the specimen's sequence
+     */
+    constructor(
+        visualizer: VisualizerInterface,
+        sequence: SequenceInterface
+    ) {
+        this.visualizer = visualizer
+        this.sequence = sequence
+    }
+    /**
+     * Call this as soon after construction as possible once the HTML
+     * element has been mounted
+     */
+    setup(location: HTMLElement) {
+        this.location = location
+        this.visualizer.view(this.sequence)
+        this.visualizer.inhabit(this.location)
+        this.visualizer.show()
+        this.isSetup = true
+    }
+    /**
+     * Returns the specimen's visualizer
+     */
+    getVisualizer(): VisualizerInterface {
+        return this.visualizer
+    }
+    /**
+     * Returns the specimen's sequence
+     */
+    getSequence(): SequenceInterface {
+        return this.sequence
+    }
+    /**
+     * Assigns a new visualizer to this specimen and updates its sequence
+     * to match the specimen. It also ensures this visualizer inhabits
+     * the correct HTML element and begins to render.
+     */
+    setVisualizer(visualizer: VisualizerInterface) {
+        this.visualizer = visualizer
+        if (this.isSetup) this.setup(this.location!)
+    }
+    /**
+     * Assigns a new sequence to this specimen and updates the visualizer
+     * to reflect this change in the render.
+     */
+    setSequence(sequence: SequenceInterface) {
+        this.sequence = sequence
+        this.visualizer.view(this.sequence)
+    }
```

Comment:
Looks like the semantics of what you are trying to do is better fit by properties 'sequence' and 'visualizer' with setter methods; that way we can avoid the ungainly `getXxx` and `setXxx` in the code that uses this class.